text/0055-remove-unspecified.md

```diff
@@ -0,0 +1,194 @@
+# Explicit Unspecified Entities
+
+## Related issues and PRs
+
+- Related RFCs: [RFC35](https://github.com/cedar-policy/rfcs/pull/35), [RFC47](https://github.com/cedar-policy/rfcs/pull/47)
+- Implementation PR(s):
+
+## Timeline
+
+- Started: 2024-02-27
+
+## Summary
+
+Cedar currently supports _unspecified entities_, which are entities of a special, unique type that have no attributes, and are not ancestors or descendants of any other entity in the store. Unspecified entities are intended to act as placeholders for entities that don’t influence authorization (see examples below).
+
+The current implementation of unspecified entities results in several corner cases in our code (especially the validator), which have been difficult to maintain (see discussion on [this recent PR](https://github.com/cedar-policy/cedar/pull/603)). The concept of unspecified entities is (in our opinion) confusing for users, and will only become more confusing once we stabilize the partial evaluation experimental feature, which allows “unknown” entities that are subtly different from “unspecified” entities.
+
+This RFC proposes to drop Cedar support for unspecified entities, instead recommending that users create their own application-specific entity types that act like unspecified entities.
+
+## Basic example
+
+Cedar uses a Principal, Action, Resource, Context \<P,A,R,C\> model for requests.
+This model can feel awkward in cases where there is not a clear principal and/or resource for an action.
+For example, consider a "getFile" action.
+The principal here should be a `User`, and the context can store information about the file to get, but it is unclear what the resource should be.
```

Comment:
I'm not following -- wouldn't the resource be the file in this case?

Reply:
Yeah, that example didn't end up making as much sense as I thought. I'll switch back to "createFile"